connector/failoverconnector/internal/state/pipeline_selector.go

```diff
@@ -31,29 +31,27 @@ func (p *PipelineSelector) handlePipelineError(idx int) {
 	}
 	doRetry := p.indexIsStable(idx)
 	p.updatePipelineIndex(idx)
-	if !doRetry {
-		return
+	if doRetry {
+		ctx, cancel := context.WithCancel(context.Background())
+		p.RS.InvokeCancel()
+		p.RS.UpdateCancelFunc(cancel)
+		p.enableRetry(ctx)
```

Comment:
This doesn't change any behavior right? Idiomatically we err towards returning early to keep the code flat.

Reply:
Ah yep meant to change that back